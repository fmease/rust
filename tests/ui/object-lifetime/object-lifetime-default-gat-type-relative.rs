trait Outer { type Ty<'a, T: 'a + ?Sized>; }
trait Inner {}

// FIXME: Ideally, we would deduce `dyn Inner + 'r` from bound `'a` on ty param `T` of assoc ty `Ty`
//        but for that we'd need to somehow obtain the resolution of the type-relative path `T::Ty`
//        from HIR ty lowering (it resolves to `<T as Outer>::Ty`).
fn f<'r, T: Outer>(x: T::Ty<'r, dyn Inner>) {}
//~^ ERROR lifetime bound for this object type cannot be deduced from context

fn main() {}
