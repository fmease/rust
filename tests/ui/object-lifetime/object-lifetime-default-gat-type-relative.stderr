error[E0228]: the lifetime bound for this object type cannot be deduced from context; please supply an explicit bound
  --> $DIR/object-lifetime-default-gat-type-relative.rs:7:33
   |
LL | fn f<'r, T: Outer>(x: T::Ty<'r, dyn Inner>) {}
   |                                 ^^^^^^^^^

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0228`.
